odo/backends/aws.py

```diff
@@ -138,6 +138,37 @@ def resource_s3_csv(uri, **kwargs):
     return S3(CSV)(uri, **kwargs)
 
 
+@resource.register('s3://.*\*.csv(\.gz)?', priority=19)
+def resource_s3_csv_glob(uri, **kwargs):
+    con = get_s3_connection()
+    result = urlparse(uri)
+    bucket = con.get_bucket(result.netloc)
+    key = result.path.lstrip('/')
+
+    all_keys = bucket.list(prefix=key.split('*')[0])
+    matched = _filter_to_matched(key, all_keys)
+    uris = [_key_to_uri(k, bucket) for k in matched]
```

Comment:
i think it's okay to inline `_filter_to_matched` and `_key_to_uri`, IMO it's pretty clear what they are doing